### DISCUSSION TITLE
How to set transmission power?

### DISCUSSION THREAD

**Discussion by RainerHeintzmann** [tbnobody/OpenDTU#1252] [General]
Ich bekomme keine Verbindung zum TSOL M350 Wechselrichter. Die DTU ist ca. 2m entfernt und die Antenne Parallel ausgerichtet.
Dazwischen ist nur eine Balkontür. Die Seriennummer (ich denke das ist wohl der Klebezettel auf dem Wechselrichter) beginnt mit 11.

In der Anleitung steht, dass man die Sendeleistung erhöhen kann, aber der Punkt "DTU Settings -> Power Level" oder irgendetwas ähnliches existiert nicht in der Software.
Ich habe den Wechselrichter noch mal aus der Software entfernt und wieder hinzugefügt. Jetzt sehe ich (wie auch voher) nur Nullen.
Und folgende Information:
"No Information available
Did not receive any valid data from the inverter till now. Still trying..."

Ist der TSOL M350 mit 11er nummer vielleicht doch nicht kompatibel? Das wäre ja schade.

**Comment by RainerHeintzmann** [comment]
.... was mich etwas stutzig macht ist folgendes unter "Info->System"

Radio Information
nRF24 Status	not configured
nRF24 Chip Status	
nRF24 Chip Type	Unknown
CMT2300A Status	not configured
CMT2300A Chip Status

**Reply by Solarteur** [reply]
Das CMT2300A-Modul wird für einen TSOL M350 nicht benötigt und bleibt in der Konfiguration außen vor.

Unter "Settings"DTU Settings" kann die "NRF24 Transmitting power" die Sendeleistung eingestellt werden. Bei mir reicht durch die Betondecke der Garage "Minimum". 

Sind unter "Settings/Device" für das NRF24-Modul Werte eingetragen - s. Screenshot.

![grafik](https://github.com/tbnobody/OpenDTU/assets/117977338/7a81e6bd-1f73-4939-87eb-8edb6e21934f)

Falls nein, fehlt nach meiner Einschätzung im "Config Management" die passende "pin_mapping.json".

Details dazu hier: https://github.com/tbnobody/OpenDTU/wiki/Device-Profiles

Ansonsten: Die NRF24-Module haben eine große Schwankungsbreite mit Blick auf die Qualität - manche sind auch von Anfang an defekt.

**Comment by RainerHeintzmann** [comment]
Danke. Es lag letztlich daran, dass ich eine schon vorhandene Firmware mit dem GitHub Repo überschrieben hatte.
Geholfen hat mir folgende Anleitung sehr:

Am einfachsten das https://github.com/tbnobody/OpenDTU/releases/download/v23.8.8/opendtu-generic.factory.bin neu Flashen.
Das kann zum Beispiel ganz einfach online mit dem WebFlash (funktioniert nur mit Chrome oder Edge) gemacht werden.
https://espressif.github.io/esptool-js/

Nach dem Verbinden am besten ein "Erase Flash" zum kompletten löschen des Moduls machen. Danach ein die Datei opendtu-generic.factory.bin auf die Adresse 0x0 flashen.
Danach sollte schon mal der Status vom NRF-Modul wieder passen.
Wie das Display aktiviert werden kann ist hier beschrieben: https://www.wib-dtu.eu/opendtu-display-re-aktivieren/
